source/camera/CullingManager.cpp

```diff
@@ -0,0 +1,289 @@
+    mWalk.myArray[3] =  Vector3i(mOgreVectorsArray[3]);
+    mWalk.buildSlopes();
+    oldWalk = mWalk;
+    oldWalk.prepareWalk();
+    mWalk.prepareWalk();
+    hideAllTiles();
+    newBashAndSplashTiles(SHOW);
+
+    mCullTilesFlag = true;
+}
+
+void CullingManager::startDebugging()
+{
+    mDebug = true;
+}
+
+void CullingManager::stopCreatureCulling()
+{
+    mCullCreaturesFlag = false;
+}
+
+void CullingManager::stopTileCulling()
+{
+    // mOldTop = mTop;
+    // mOldBottom = mBottom;
+    // mOldMiddleLeft = mMiddleLeft;
+    // mOldMiddleRight = mMiddleRight;
+
+    oldWalk = mWalk;
+    newBashAndSplashTiles(HIDE);
+    showAllTiles();
+    mCullTilesFlag = false;
+}
+
+void CullingManager::hideAllTiles(void)
+{
+    GameMap* gm = mCm->mGameMap;
+    if (!gm)
+        return;
+
+    for (int jj = 0; jj < gm->getMapSizeY() ; ++jj)
+    {
+        for (int ii = 0; ii < gm->getMapSizeX(); ++ii)
+        {
+            gm->getTile(ii,jj)->hide();
+        }
+    }
+}
+
+
+
+void CullingManager::showAllTiles(void)
+{
+    GameMap* gm = mCm->mGameMap;
+    if (!gm)
+        return;
+
+    for (int jj = 0; jj < gm->getMapSizeY() ; ++jj)
+    {
+        for (int ii = 0; ii < gm->getMapSizeX(); ++ii)
+        {
+            gm->getTile(ii,jj)->show();
+        }
+    }
+}
+
+
+
+
+
+void CullingManager::newBashAndSplashTiles(int64_t mode){
+    int64_t xxLeftOld = oldWalk.getTopLeftVertex().x;
+    int64_t xxRightOld= oldWalk.getTopRightVertex().x;
+    int64_t xxLeft = mWalk.getTopLeftVertex().x;
+    int64_t xxRight= mWalk.getTopRightVertex().x;
+    int64_t DxRight, DxLeft;
+    int64_t xxp, yyp;
+    std::stringstream ss;
+    int64_t bb = (( std::min(mWalk.getBottomLeftVertex().y , oldWalk.getBottomRightVertex().y) >> mPrecisionDigits) - 2) << mPrecisionDigits;
+
+    for (int64_t yy = ((std::max(mWalk.getTopLeftVertex().y , oldWalk.getTopRightVertex().y  ) >> mPrecisionDigits) + 2) << mPrecisionDigits;  yy >= bb; yy -= Unit)
+    {
+        oldWalk.notifyOnMoveDown(yy);
+        mWalk.notifyOnMoveDown(yy);
+        DxLeft = mWalk.getCurrentXLeft(yy);
+        xxLeft = DxLeft;
+        xxLeftOld = oldWalk.getCurrentXLeft(yy);
+        DxRight = mWalk.getCurrentXRight(yy);
+        xxRight = DxRight;
+        xxRightOld = oldWalk.getCurrentXRight(yy);
+   
+        if(mDebug)
+        {
+            ss << endl;
+            ss << "DxLeft, DxRight: " << double(DxLeft)/Unit << " " << double(DxRight)/Unit << " " ;
+            LogManager::getSingleton().logMessage( ss.str(),LogMessageLevel::NORMAL);
+            ss.str("");
+            ss << "xxLeft " << (xxLeft  >> mPrecisionDigits) << "xxRight " << (xxRight  >> mPrecisionDigits );
+            LogManager::getSingleton().logMessage( ss.str(),LogMessageLevel::NORMAL);
+            ss.str("");
+        }
+    
+        int64_t mm = ((std::min(xxLeft, xxLeftOld) >> mPrecisionDigits) << mPrecisionDigits) ;
+        if(std::min(xxLeft, xxLeftOld) < max(xxRight,xxRightOld) )
```

Comment:
Please never ever do multiline ifs without brackets. The if clearly needs brackets in similar cases, and in fact, in this case, it should be inverted with a return to avoid useless cyclomatic complexity:
```c++
if(std::min(xxLeft, xxLeftOld) >= std::max(xxRight,xxRightOld))
    return;
```